lib/rules/no-fallthrough.js

```diff
@@ -90,6 +90,8 @@ module.exports = {
     meta: {
         type: "problem",
 
+        defaultOptions: [{}],
```

Comment:
https://github.com/eslint/eslint/blob/b8fb57256103b908712302ccd508f464eff1c9dc/lib/rules/no-fallthrough.js#L17

Similar to `default-case`. Note the `i` there, if a config passes in a string then the `i` is omitted. This would be a separate (technically breaking, I think) change for the rule to normalize.

Reply:
I'm confused here too. If I pass another config value (be it a string or other regex) then I expect default value no longer apply.

Isn't it safe to set `defaultOptions.commentPattern` to the value of `DEFAULT_FALLTHROUGH_COMMENT`?

Reply:
I tried it out and a unit test failed. So I think that's something that'll have to be treated as a separate issue.

IMO this PR doesn't have to switch _every_ rule to using `meta.defaultOptions`. Some would be a breaking change like this. Others literally can't per the current ESLint test suite, such as `id-length`'s `max` having a non-JSON-serializable `Infinity`.